pkg/utils/experimentHelpers.go

```diff
@@ -17,18 +17,31 @@ func (expDetails *ExperimentDetails) SetValueFromChaosExperiment(clients ClientS
 }
 
 //SetENV sets ENV values in experimentDetails struct.
-func (expDetails *ExperimentDetails) SetENV(engineDetails EngineDetails, clients ClientSets) {
+func (expDetails *ExperimentDetails) SetENV(engineDetails *EngineDetails, clients ClientSets) error {
 	// Get the Default ENV's from ChaosExperiment
 	log.Infoln("Getting the ENV Variables")
-	expDetails.SetDefaultEnv(clients)
+	err := expDetails.SetDefaultEnv(clients)
+	if err != nil {
+		return err
+	}
 	// OverWriting the Defaults Varibles from the ChaosEngine ENV
-	expDetails.SetEnvFromEngine(engineDetails.Name, clients)
+	err = expDetails.SetEnvFromEngine(engineDetails.Name, clients)
+	if err != nil {
+		return err
+	}
+	// Get engineUID from the chaos-runner's label
+	err = SetEngineUID(engineDetails, clients)
+	if err != nil {
+		return err
+	}
 	// Adding some addition ENV's from spec.AppInfo of ChaosEngine
 	expDetails.Env["CHAOSENGINE"] = engineDetails.Name
 	expDetails.Env["APP_LABEL"] = engineDetails.AppLabel
 	expDetails.Env["APP_NAMESPACE"] = engineDetails.AppNamespace
 	expDetails.Env["APP_KIND"] = engineDetails.AppKind
 	expDetails.Env["AUXILIARY_APPINFO"] = engineDetails.AuxiliaryAppInfo
+	expDetails.Env["ENGINE_UID"] = engineDetails.UID
```

Comment:
can this be made generic? CHAOS_UID (there may be non-engine parents of the job too?)